src/interpreter/library/mod.rs

```diff
@@ -117,37 +121,35 @@ impl StandardLibrarySearcher {
     }
 }
 
-impl LibrarySearcher for StandardLibrarySearcher {
-    fn search_lib(&self, library_name: Located<LibraryName>) -> Option<PathBuf> {
-        match std::env::current_dir() {
-            Ok(pwd) => Some(
-                pwd.join("lib")
-                    .join(library_name.extract_data().path())
-                    .with_extension("sld"),
-            ),
-            Err(_) => None,
-        }
+impl StandardLibrarySearcher {
+    pub fn get_include_library(library_name: &LibraryName) -> Option<&'static str> {
+        let defined_librarys = include::library_strs()
+            .into_iter()
+            .collect::<HashMap<_, _>>();
```

Comment:
seems this collecting can be written in library_strs

Reply:
I think that library_strs is just a list of include library, not a hashmap, usage in this function required a hash search, but in other place may just need to iterate over whole list

Reply:
So, in this single function, if you just wanna get lib from name over a vector, you should search it or binary search it from a sorted one.

Reply:
Constructing a HashMap is an O(n)  complexity step with a big constant factor.